private/bufpkg/bufapimodule/module_reader.go

```diff
@@ -80,8 +78,10 @@ func (m *moduleReader) GetModule(ctx context.Context, modulePin bufmoduleref.Mod
 	if err != nil {
 		return nil, err
 	}
-	if err := warnIfDeprecated(ctx, m.repositoryClientFactory, modulePin, m.logger); err != nil {
-		return nil, err
+	if logger, repositoryClientFactory := m.logger, m.repositoryClientFactory; logger != nil && repositoryClientFactory != nil {
```

Comment:
Couldn't this simply be `if m.logger != nil && m.repositoryClientFactory != nil`? I don't see the need to assign those to additional variables here just to do a nil check.

Reply:
resolved in [33582fe](https://github.com/bufbuild/buf/pull/2594/commits/33582fe8cd7535e86eefdcf5c4a4e8dad71a2374)